### PR TITLE
rel: Prepare v0.10.0 release

### PR DESCRIPTION
**Pull request by jharley**
## Which problem is this PR solving?
Prepares the next release of this library.

## Short description of the changes

- Update version to v0.10.0
- Add changelog entry (and missing one for v0.9.0)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,34 @@
...
+- maint(deps): bump the otel-dependencies group with 1 update (#186)
+- maint(deps): bump go.opentelemetry.io/otel from 1.21.0 to 1.23.0 in /examples/baggage (#187)
+- maint(deps): bump go.opentelemetry.io/otel from 1.21.0 to 1.23.0 (#185)
+- maint(deps): bump github.com/honeycombio/otel-config-go from 1.13.0 to 1.13.1 in /examples/baggage (#180)
+- maint(deps): bump github.com/honeycombio/otel-config-go from 1.13.0 to 1.13.1 in /examples/webhook-listener-triggers (#179)
+- maint(deps): bump github.com/honeycombio/otel-config-go from 1.13.0 to 1.13.1 (#178)
+- maint: update codeowners to pipeline-team (#177)
+- maint: update codeowners to pipeline (#176)
+
+## v0.9.0 (2023-11-23)
```
Comment (JamieDanielson) [inline CHANGELOG.md:21]: Oh damn look at me failing to update the changelog in the last release on #175  😬  Thank you!

**Review by JamieDanielson — COMMENTED** [review]
Can you also update the README with the current versions this is built with for otel and otel-config-go? Looks like it should be otel 1.23/v0.48 and otel-config-go 1.13.1?

**Comment by MikeGoldsmith** [comment]
> Can you also update the README with the current versions this is built with for otel and otel-config-go? Looks like it should be otel 1.23/v0.48 and otel-config-go 1.13.1?

Fixed 👍🏻

**Review by MikeGoldsmith — APPROVED** [review]

**Comment by jharley** [comment]
Thanks both! 🥳

**Pull request merged**